nixos/modules/services/x11/xbanish.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.xbanish;
+
+in {
+  options.services.xbanish = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Enable xbanish to hide your mouse pointer while not in use";
+    };
+
+    arguments = mkOption {
+      description = "Arguments to pass to xbanish command";
+      default = "";
+      example = "-d -i shift";
+      type = types.str;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.xbanish = {
+      description = "xbanish";
+      wantedBy = [ "default.target" ];
```

Comment:
The description is a bit sparse... And for `wantedBy` I think `graphical.target` is a better fit